### PR TITLE
[settings] Initial migration to Shoelace

### PR DESCRIPTION
**Pull request by erikinkinen**
This commit migrates settings app to use Shoelace.

### CODE REVIEW

File: apps/settings/index.html [hunk 1]
```diff
@@ -17,21 +17,77 @@
...
+      <header>
+        <sl-input id="search" clearable data-l10n-id="search-settings" data-l10n-attrs="placeholder">
+          <sl-icon name="search" slot="prefix" />
+        </sl-input>
+      </header>
+      <sl-menu id="main-menu">
+        <sl-menu-label>Connectivity</sl-menu-label>
+        <sl-menu-item value="wifi">
+            <sl-icon name="wifi" slot="prefix"></sl-icon>
+            <sl-icon name="chevron-right" slot="suffix"></sl-icon>
```
Comment (fabricedesre) [inline apps/settings/index.html:37]: Not much we can do now, but it looks like Shoelace will provide that as a slot in a future release: https://github.com/shoelace-style/shoelace/blob/8ae987ea691fed9faa1906b88964cc5932809cba/src/components/menu-item/menu-item.ts#L79

File: apps/settings/index.html [hunk 2]
```diff
@@ -17,21 +17,77 @@
...
+            <span data-l10n-id="mobile"></span>
+        </sl-menu-item>
+        <sl-menu-item value="bluetooth">
+            <sl-icon name="bluetooth" slot="prefix"></sl-icon>
+            <sl-icon name="chevron-right" slot="suffix"></sl-icon>
+            <span data-l10n-id="bluetooth"></span>
+        </sl-menu-item>
+        <sl-divider></sl-divider>
+        <sl-menu-label>Device</sl-menu-label>
```
Comment (fabricedesre) [inline apps/settings/index.html:51]: l10n

File: apps/settings/index.html [hunk 3]
```diff
@@ -17,21 +17,77 @@
...
+            <sl-icon name="chevron-right" slot="suffix"></sl-icon>
+            <span data-l10n-id="sound"></span>
+        </sl-menu-item>
+        <sl-menu-item value="security">
+            <sl-icon name="lock" slot="prefix"></sl-icon>
+            <sl-icon name="chevron-right" slot="suffix"></sl-icon>
+            <span data-l10n-id="security"></span>
+        </sl-menu-item>
+        <sl-divider></sl-divider>
+        <sl-menu-label>Information</sl-menu-label>
```
Comment (fabricedesre) [inline apps/settings/index.html:83]: l10n

File: apps/settings/js/bootstrap.js [hunk 4]
```diff
@@ -1,41 +1,78 @@
...
+  let sections = this.document.querySelectorAll("section");
+  sections.forEach(i => {
+    i.classList.add("hidden");
+  })
+  if (window.location.hash === "") {
+    elem("main").classList.remove("hidden");
+  } else {
+    let name = window.location.hash.substring(1);
+    elem("spinner").classList.remove("hidden");
+    let e = elem(name);
```
Comment (fabricedesre) [inline apps/settings/js/bootstrap.js:17]: nit: please avoid single letter variables.

What happens if the hash does not match an existing panel? We should fallback on the main screen.

File: apps/settings/js/bootstrap.js [hunk 5]
```diff
@@ -1,41 +1,78 @@
...
+    .then((response) => response.text())
+    .then((html) => {
+        e.innerHTML = html;
+        elem("spinner").classList.add("hidden");
+        e.dispatchEvent(bootstrapEvent);
+        e.classList.remove("hidden");
+    })
+    .catch((error) => {
+        console.error(error);
+    });
```
Comment (fabricedesre) [inline apps/settings/js/bootstrap.js:30]: nit: that looks similar to just `.catch(console.error);`

File: apps/settings/js/bootstrap.js [hunk 6]
```diff
@@ -1,41 +1,78 @@
...
+
+    await import(`/panels/${name}/js/bootstrap.js`);
+    fetch(`/panels/${name}/index.html`)
+    .then((response) => response.text())
+    .then((html) => {
+        e.innerHTML = html;
+        elem("spinner").classList.add("hidden");
+        e.dispatchEvent(bootstrapEvent);
+        e.classList.remove("hidden");
+    })
```
Comment (fabricedesre) [inline apps/settings/js/bootstrap.js:27]: if you go back and forth in the same panel, won't that reload it each time?

File: apps/settings/index.html [hunk 7]
```diff
@@ -17,21 +17,77 @@
...
-        <panel-link name="wifi" icon="wifi"></panel-link>
-        <panel-link name="telephony" icon="phone"></panel-link>
-        <panel-link name="display" icon="image" disabled></panel-link>
-        <panel-link name="lockscreen" icon="lock"></panel-link>
-        <panel-link name="region_language" icon="globe" disabled></panel-link>
-        <panel-link name="date_time" icon="clock" disabled></panel-link>
-        <panel-link name="system_info" icon="info"></panel-link>
-      </ul>
-    </div>
+    <section id="spinner" style="display: flex; align-items: center; justify-content: center;">
```
[the remark this answers is not in the document]
Comment (erikinkinen) [reply apps/settings/index.html]: They were effective but I moved them to the css file in any case.

File: apps/settings/index.html [hunk 8]
```diff
@@ -17,21 +17,77 @@
...
+      <sl-spinner style="font-size: 4rem;"></sl-spinner>
+    </section>
+    <section id="main" class="hidden">
+      <header>
+        <sl-input id="search" clearable data-l10n-id="search-settings" data-l10n-attrs="placeholder">
+          <sl-icon name="search" slot="prefix" />
+        </sl-input>
+      </header>
+      <sl-menu id="main-menu">
+        <sl-menu-label>Connectivity</sl-menu-label>
```
[the remark this answers is not in the document]
Comment (erikinkinen) [reply apps/settings/index.html]: Oops, forgot that. Should be fixed now.

File: apps/settings/index.html (hunk 1 above)
Comment (erikinkinen) [reply apps/settings/index.html:37]: Until then, we'll do it like this, I guess.

File: apps/settings/js/bootstrap.js (hunk 4 above)
Comment (erikinkinen) [reply apps/settings/js/bootstrap.js]: Replaced `e` with `panel` and implemented a fall-back to main menu.

File: apps/settings/js/bootstrap.js (hunk 5 above)
Comment (erikinkinen) [reply apps/settings/js/bootstrap.js]: Oh, yeah, fixed!

File: apps/settings/js/bootstrap.js (hunk 6 above)
Comment (erikinkinen) [reply apps/settings/js/bootstrap.js]: I was aware of that already and I pushed my fix for that now.

**Comment by fabricedesre** [comment]
Closing since this was done in another commit.

**Pull request closed without merging**